dlt/destinations/impl/sink/configuration.py

```diff
@@ -0,0 +1,71 @@
+from typing import TYPE_CHECKING, Optional, Final, Callable, Union, Any
+from importlib import import_module
+
+from dlt.common.configuration import configspec
+from dlt.common.destination import TLoaderFileFormat
+from dlt.common.destination.reference import (
+    DestinationClientConfiguration,
+    CredentialsConfiguration,
+)
+from dlt.common.typing import TDataItems
+from dlt.common.schema import TTableSchema
+from dlt.common.configuration.exceptions import ConfigurationValueError
+
+
+TSinkCallable = Callable[[Union[TDataItems, str], TTableSchema], None]
+
+
+@configspec
+class SinkClientCredentials(CredentialsConfiguration):
+    callable: Optional[str] = None  # noqa: A003
+
+    def parse_native_representation(self, native_value: Any) -> None:
+        # a callable was passed in
+        if callable(native_value):
+            self.resolved_callable: TSinkCallable = native_value
+        # a path to a callable was passed in
+        if isinstance(native_value, str):
+            self.callable = native_value
+
+    def to_native_representation(self) -> Any:
+        return self.resolved_callable
+
+    def on_resolved(self) -> None:
+        if self.callable:
+            try:
+                module_path, attr_name = self.callable.rsplit(".", 1)
+                dest_module = import_module(module_path)
+            except ModuleNotFoundError as e:
+                raise ConfigurationValueError(
+                    f"Could not find callable module at {module_path}"
+                ) from e
+            try:
+                self.resolved_callable = getattr(dest_module, attr_name)
+            except AttributeError as e:
+                raise ConfigurationValueError(
+                    f"Could not find callable function at {self.callable}"
+                ) from e
+
+        if not hasattr(self, "resolved_callable"):
+            raise ConfigurationValueError("Please specify callable for sink destination.")
+
+        if not callable(self.resolved_callable):
+            raise ConfigurationValueError("Resolved Sink destination callable is not a callable.")
+
+
+@configspec
+class SinkClientConfiguration(DestinationClientConfiguration):
+    destination_type: Final[str] = "sink"  # type: ignore
```

Comment:
mhhhh I think we'll abuse this config and use the name of the destination as type?

Reply:
yes, this is the way it works already (see the config/secrets tests), unless I misunderstand what you are saying.